ToolBox/Benzinga/BenzingaNewsDataConverter.cs

```diff
@@ -0,0 +1,451 @@
+
+        // FIXME: Associated issue: https://github.com/QuantConnect/Lean/issues/3849
+        private readonly HashSet<string> _forbiddenTickers = new HashSet<string>()
+        {
+            "CON",
+            "PRN",
+            "AUX",
+            "NUL",
+            "COM1",
+            "COM2",
+            "COM3",
+            "COM4",
+            "COM5",
+            "COM6",
+            "COM7",
+            "COM8",
+            "COM9",
+            "LPT1",
+            "LPT2",
+            "LPT3",
+            "LPT4",
+            "LPT5",
+            "LPT6",
+            "LPT7",
+            "LPT8",
+            "LPT9"
+        };
+
+        /// <summary>
+        /// Creates a new instance of the Benzinga converter
+        /// </summary>
+        /// <param name="sourceDirectory">Directory to read data from. This should be the Benzinga data folder (e.g. /Data/alternative/benzinga)</param>
+        /// <param name="destinationDirectory">Directory to write data to. This should be the final Benzinga data folder (e.g. /Data/alternative/benzinga)</param>
+        public BenzingaNewsDataConverter(DirectoryInfo sourceDirectory, DirectoryInfo destinationDirectory, DirectoryInfo processedFilesDirectory)
+        {
+            _sourceDirectory = sourceDirectory;
+            _destinationDirectory = destinationDirectory;
+            _processedFilesDirectory = processedFilesDirectory;
+            _destinationDirectory.Create();
+            _isWindows = OS.IsWindows;
+        }
+
+        public bool Convert(DateTime date)
+        {
+            var newsArticles = new DirectoryInfo(
+                Path.Combine(
+                    _sourceDirectory.FullName,
+                    $"{date:yyyyMMdd}"
+                )
+            );
+
+            if (!newsArticles.Exists)
+            {
+                Log.Error($"BenzingaDataConverter.Convert(): Data not found for date {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            try
+            {
+                WriteToFile(Process(newsArticles));
+            }
+            catch (Exception error)
+            {
+                Log.Error(error, $"Failed to complete processing of Benzinga news data for {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Convert the raw data to BenzingaNews instances.
+        /// </summary>
+        /// <param name="dateDirectory">Directory for a given date, e.g. _sourceDirectory/20181227/</param>
+        /// <returns>Enumerable of <see cref="BenzingaNews"/> instances</returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.json", SearchOption.TopDirectoryOnly))
+            {
+                foreach (var article in JsonConvert.DeserializeObject<JArray>(File.ReadAllText(publication.FullName)))
```

Comment:
Wondering if you could directly deserialize into a `List<BenzingaNews>`, maybe worth a test

Reply:
It works, but we'd lose logging from the converter as a result 😢